tools/tpm2_create.c

```diff
@@ -220,7 +220,7 @@ int create(TSS2_SYS_CONTEXT *sapi_context)
     }
 
     if (ctx.flags.O) {
-        bool res = files_save_bytes_to_file(ctx.opr_path, outPrivate.t.buffer, outPrivate.t.size);
+        bool res = files_save_private(ctx.opr_path, &outPrivate);
```

Comment:
I would just include this part and the others that fix the issue and avoid the other bits that deal with the wrong size. This was clearly a bug so I would just fix it. Although the others changes cause no harm so up to you.